driver-async/src/main/com/mongodb/async/client/ClientSessionImpl.java

```diff
@@ -44,19 +50,27 @@
 
     @Override
     public boolean hasActiveTransaction() {
-        return inTransaction;
+        return transactionState == TransactionState.IN || (transactionState == TransactionState.DONE && commitInProgress);
     }
 
     @Override
     public boolean notifyMessageSent() {
-        boolean firstMessage = !messageSent;
-        messageSent = true;
-        return firstMessage;
+        if (transactionState == TransactionState.DONE && !commitInProgress) {
```

Comment:
This is to satisfy the spec requirement that the first operation executed after a transaction commits, but not in a new transaction, moves the state from DONE to NONE